### PR TITLE
Feature request: Limit the database type. #227

### LINKED ISSUE #227: Limit the database type.
We use MySQL in our dev/test/production environment, and use H2 in unit test environment.

Configuration is in src/main/resources/datasource.xml and src/test/resources/datasource.xml.

If the developer rename the src/test/resources/datasource.xml config file, the MySQL datasource will be used in unit test environment. And if the developer run unit tests, the @DataSet will clear the data in MySQL database.

So I wonder to know how to limit the database type for database-rider or dbunit in unit test environment? In this way, We don't need to worry the data in MySQL database.

### PR DESCRIPTION
**Pull request by kerraway**
Add "expectedDbType" config option for DBUnit and dbunit.yml.

In the process of initialization, if the actual database type is different from the expected database type, exception will be thrown unless the expected database type is `UNKNOWN`. Default is `UNKNOWN`.

Close #227 .

### CODE REVIEW

**Comment by kerraway** [comment]
Add unit test for RiderDataSource to increase the coverage.

> [Coverage Status](https://coveralls.io/builds/32295375)
> 
> Coverage decreased (-0.01%) to 81.755% when pulling **[d30a284](https://github.com/database-rider/database-rider/commit/d30a28435a53426146221324c87a7bf92d9766e1) on kerraway:limit-db-type** into **[3927d12](https://github.com/database-rider/database-rider/commit/3927d122b3dd723f9e49f9e30063e635c9e49a21) on database-rider:master**.

Add unit test for RiderDataSource to increase the coverage.

File: README.adoc [hunk 1]
```diff
@@ -342,6 +342,7 @@ leakHunter: false
 mergeDataSets: false
 caseInsensitiveStrategy: !!com.github.database.rider.core.api.configuration.Orthography 'UPPERCASE' <1>
 raiseExceptionOnCleanUp: false <2>
+expectedDbType: HSQLDB <3>
```
Comment (rmpestano) [inline README.adoc:345]: I think here the default is UNKNOWN, right? is this doc we should put the default values

Comment (kerraway) [reply README.adoc:345]: OK, I modify it to UNKNOWN.

**Comment by kerraway** [comment]
> Awesome, thank you very much!

By the way, When do you publish 1.16.0 jar?

**Comment by rmpestano** [comment]
> By the way, When do you publish 1.16.0 jar?

Probably this weekend

**Pull request merged**